lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/LocationDaoImpl.java

```diff
@@ -12,11 +14,14 @@
  * @author Jakub Polacek
  */
 
+@Repository
+@Transactional
 public class LocationDaoImpl implements LocationDao {
 
-    @PersistenceContext(unitName = "location-unit", type = PersistenceContextType.EXTENDED)
+    @PersistenceContext
     private EntityManager em;
 
+    @Override
     public void addLocation(Location location) throws IllegalArgumentException {
```

Comment:
Can you add an if for when location.getId() != null to throw an exception? Or is there a use case when we don't want to do that?

Reply:
Jakub to spravil.